overlord/snapstate/handlers_components_download_test.go

```diff
@@ -0,0 +1,172 @@
+package snapstate_test
+
+import (
+	"context"
+	"errors"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/overlord/auth"
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/snapstate/snapstatetest"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/progress"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/snap/naming"
+	"github.com/snapcore/snapd/snap/snaptest"
+	"github.com/snapcore/snapd/store"
+	"github.com/snapcore/snapd/store/storetest"
+
+	. "gopkg.in/check.v1"
+)
+
+type downloadComponentSuite struct {
+	baseHandlerSuite
+
+	fakeStore *fakeStore
+}
+
+var _ = Suite(&downloadComponentSuite{})
```

Comment:
These tests target download-component specifically, is there (or will there be) another test that tests this with a more complete change? That would cover whether DownloadInfo is unexpectedly nil and things like that

Reply:
Yes, more complete tests are coming once we can actually use this handler.